### PR TITLE
Fix some bugs

### PR DESCRIPTION
**Pull request by danwinkler**
- fix classify_v2 generator bug
- add malloc checks, convert alloca->malloc in sigprocv2
- hack to fix zplots.im stack

VSCode also took it upon itself to auto format every c file I touched.

### CODE REVIEW

**Review by zack-erisyon — CHANGES_REQUESTED** [review]
Ah! No! I need my K&R braces back. Can you get the editor to un-fuck this?

File: plaster/run/sigproc_v2/c_radiometry/radiometry.c [hunk 1]
```diff
@@ -243,12 +235,16 @@ char *radiometry_field_stack_one_peak(RadiometryContext *ctx, Index peak_i) {
     // Shape
     Index n_divs_minus_one = ctx->n_divs - 1;
 
-    Float64 *psf_pixels = (Float64 *)alloca(sizeof(Float64) * mea_sq);
-    Float64 *dat_pixels = (Float64 *)alloca(sizeof(Float64) * mea_sq);
-    Float64 *msk_pixels = (Float64 *)alloca(sizeof(Float64) * mea_sq);
+    Float64 *psf_pixels = (Float64 *)malloc(sizeof(Float64) * mea_sq);
```
Comment (danwinkler) [inline plaster/run/sigproc_v2/c_radiometry/radiometry.c:238]: code change

File: plaster/run/sigproc_v2/c_radiometry/radiometry.c [hunk 2]
```diff
@@ -345,10 +337,13 @@ char *radiometry_field_stack_one_peak(RadiometryContext *ctx, Index peak_i) {
         out[3] = asr;
     }
 
+    free(psf_pixels);
```
Comment (danwinkler) [inline plaster/run/sigproc_v2/c_radiometry/radiometry.c:340]: code change

File: plaster/run/sigproc_v2/c_sub_pixel_align/sub_pixel_align.c [hunk 3]
```diff
@@ -163,8 +138,12 @@ char *sub_pixel_align_one_cycle(SubPixelAlignContext *ctx, Index cy_i) {
     Float64 *slice_buffer = (Float64 *)malloc(sizeof(Float64) * width);
     Float64 *large_slice_buffer = (Float64 *)malloc(sizeof(Float64) * large_width);
 
+    ensure(offset_samples != NULL, "malloc failed");
```
Comment (danwinkler) [inline plaster/run/sigproc_v2/c_sub_pixel_align/sub_pixel_align.c:141]: code change

File: plaster/run/sigproc_v2/c_sub_pixel_align/sub_pixel_align.c [hunk 4]
```diff
@@ -205,11 +183,12 @@ char *context_init(SubPixelAlignContext *ctx) {
     ctx->_n_slices = n_slices;
 
     Float64 *slice_buffer = (Float64 *)malloc(sizeof(Float64) * width);
-    Size cy0_slices_shape[2] = { n_slices, scale * width };
+    ensure(slice_buffer != NULL, "malloc failed");
```
Comment (danwinkler) [inline plaster/run/sigproc_v2/c_sub_pixel_align/sub_pixel_align.c:186]: code change

File: plaster/tools/c_common/c_common.c [hunk 5]
```diff
@@ -469,51 +479,37 @@ F64Arr f64arr(void *base, Size n_dims, Size *shape) {
...
 
-
 F64Arr f64arr_malloc(Size n_dims, Size *shape) {
     Size size = 1;
-    for(Index i=0; i<n_dims; i++) {
+    for(Index i = 0; i < n_dims; i++) {
         size *= shape[i];
     }
     Float64 *buffer = (Float64 *)malloc(sizeof(Float64) * size);
+    ensure(buffer != NULL, "malloc failed");
```
Comment (danwinkler) [inline plaster/tools/c_common/c_common.c:495]: code change

**Comment by danwinkler** [comment]
@zack-erisyon I'm in general a believer in using auto code formatters everywhere, so I took this as an opportunity to try and encode your style for clang-format. I didn't get it perfect, so let me know if there's anything in particular that bugs you about this formatting style.

I've also marked the places where actual code changes occur in c code to make it easier to review.

**Review by zack-erisyon — APPROVED** [review]

**Pull request merged**